python/ray/serve/controller.py

```diff
@@ -220,10 +220,7 @@ def _put_serve_snapshot(self) -> None:
             entry["class_name"] = (
                 deployment_info.replica_config.func_or_class_name)
             entry["version"] = deployment_info.version or "None"
-            # TODO(architkulkarni): When we add the feature to allow
-            # deployments with no HTTP route, update the below line.
-            # Or refactor the route_prefix logic in the Deployment class.
-            entry["http_route"] = route_prefix or f"/{deployment_name}"
+            entry["http_route"] = route_prefix
```

Comment:
I am not entirely sure about this part. @architkulkarni, does this look right to you?